core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java

```diff
@@ -326,9 +326,13 @@ void validatePatchTransactionFields(Sep31Transaction txn, Sep31PatchTransactionR
     }
   }
 
-  void preValidateQuote() throws AnchorException {
+  void preValidateQuote(boolean isQuotesRequired) throws AnchorException {
     Sep31PostTransactionRequest request = Context.get().getRequest();
 
+    if (isQuotesRequired && request.getQuoteId() == null) {
+      throw new BadRequestException("quotes_required is set to true; quote id cannot be empty");
+    }
+
     // Check if quote is provided.
     if (request.getQuoteId() == null) {
```

Comment:
👀  Can you also add a condition where `quoteSupported == false` also returns early?

```suggestion
    if (!quoteSupported || request.getQuoteId() == null) {
```

Reply:
Also, could you plz add a test for this new condition?